pulseapi/users/adapter.py

```diff
@@ -82,10 +82,12 @@ class PulseSocialAccountAdapter(DefaultSocialAccountAdapter):
     def populate_user(self, request, sociallogin, data):
         user = super().populate_user(request, sociallogin, data)
         name = data.get('name')
-        user.name = ' '.join([
-            data.get('first_name', ''),
-            data.get('last_name', '')
-        ]).strip() if not name else name
+        user.name = ' '.join(
```

Comment:
we probably want to shortcut this the other way:
```
user.name = name
if not user.name: 
  user.name = f'{data.get('first_name",'')} {data.get('last_name', '')}'.strip()
```

Reply:
Although of course the notion that a name is composed of a first/last name at all is _exceptionally_ flawed, and reconstituting it with a space in between is equally and independently flawed. Works (often but not always) for western names, breaks very quickly outside of that cultural context =D

Reply:
the issue isn't about joining but that we get `None` values in the result so using an `f` string will not fix that, we need to filter it out.

Reply:
If `data.get('first_name', '')` can yield `None`, then we should avoid using `.get()`, as the idea is that if there's data, it's real data, otherwise it's an empty string.

If we can't set a policy on the super class that deletes keys with None values, so that `.get()` will do what it's supposed to, let's use a plain get without fallback value, and a formatted string that because it's breaking an expected pattern. Rather than filtering, can we use something like this?

```
        if not name:
            first = data.get('first_name')
            last = data.get('last_name')
            full_name =  f"{first if first else ''} {last if last else ''}".strip()
            user.name = full_name if full_name else ???
```

Where that `???` is kind of an open question: clearly it's _possible_ that there's no name, and no first/last name either, if `None` is an allowed data value, so we probably do want that ultimate fallback value in there (even if it's almost certainly never getting used)